Added some reasonable defaults to LIRC's module

alex-ameen opened the pull request:
###### Motivation for this change

After spending a few hours trying to install LIRC I realized that many of my issues were from a lack of default settings that could get the program up and running. This will allow many users to let LIRC run out of the box.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

alex-ameen committed:
Added some reasonable defaults to LIRC's module

Mic92 commented on nixos/modules/services/hardware/lirc.nix:
```diff
@@ -19,12 +19,29 @@ in {
+
+          [lircmd]
+          uinput          = False
+          nodaemon        = False
```
To bad this was not added in the first place. I would prefer having sane default options. that can be extended by `extraOptions`.

Infinisil replied:
I think for now this is alright, can always add that later

Mic92 replied:
agreed.

Infinisil commented:
@BadDecisionsAlex We should strive to keep the default config as small as possible. If @ck3d's config works then that is preferred.

BadDecisionsAlex commented:
> My configuration file looks as follows:
> 
> ```
> [lircd]
> driver = ftdi
> device = product=0x6015
> ```
> I set none of the default values in my configuration file without any problems. Could you please specify which problems you have?

Sorry for the late reply, I've been swamped.
Without the default settings that I added LIRC was unable to access any of the drivers which it is normally packaged with.

alex-ameen committed:
Merge branch 'master' of https://github.com/nixos/nixpkgs

alex-ameen committed:
rolling back lirc so it does not conflict with nixpkgs master

ck3d commented:
No problem.
To check that lircd find its drivers, call following command:
```
lircd -H help
```
In my case, everything is found. The default, ftdi and 30 more drivers.

I think the main reason is the missing device setting. Please give following config a try:
```
[lircd]
device = /dev/lirc0
```
or set it as extraArguments:
extraArguments = [ "--device=/dev/lirc0" ];

I had a look into lirc sources and it showed that the default device of the lircd default driver is /dev/lirc/0 (defined as LIRC_DRIVER_DEVICE). If /dev/lirc0 is a better default, we should think about patching lirc.

As alternative we could provide two NixOS options for driver and device.

alex-ameen committed:
Merge branch 'master' of https://github.com/nixos/nixpkgs

alex-ameen committed:
genymotion: refactor src url

Fixed revisions:
1) Uses `pname`.
2) URL uses version variable in path. Data type for `url` was changed from path to string.

alex-ameen committed:
genymotion: refactor src url

The last commit mistakenly reverted the URL to "old" again. The change has been re-applied.

BadDecisionsAlex commented:
Sorry I did not mean to revive this. This has been sitting without interest for a while. I'm closing it.
